src/util/CrosisError.ts

```diff
@@ -0,0 +1,22 @@
+export default class CrosisError extends Error {
+  /** extras - a map of event data in a format appropriate to send to an error logging service */
+  public readonly extras: Record<string, unknown> | null;
+  /** tags - a map of tags to their values in a format appropriate to send to an error logging service */
+  public readonly tags: Record<string, string> | null;
+
+  constructor(
+    message: string,
+
+    extras: Record<string, unknown> | null = null,
+    tags: Record<string, string> | null = null,
+  ) {
+    super(message);
+
+    this.extras = extras;
+    this.tags = tags;
+  }
+
+  static fromError(error: Error): CrosisError {
+    return new CrosisError(error.message);
+  }
```

Comment:
```suggestion
  static fromError(error: Error): CrosisError {
    const crosisError = new CrosisError(error.message);
    crosisError.stack = error.stack;
    return crosisError
  }
```

we need to preserve stack traces :scream:

Reply:
I guess, I think all this does is drop the `fromError` part of the stack trace (everything prior will be in the new stack), but that is probably helpful for Sentry.